src/dialect.rs

```diff
@@ -2,6 +2,26 @@ use crate::allocator::{Allocator, NodePtr};
 use crate::cost::Cost;
 use crate::reduction::Response;
 
+// division with negative numbers are disallowed
+pub const NO_NEG_DIV: u32 = 1;
+
+// in conditions output, integers must use canonical encoding (i.e. no redundant
+// leading zeros)
+pub const COND_CANON_INTS: u32 = 2;
+
+// unknown condition codes are disallowed
+pub const NO_UNKNOWN_CONDS: u32 = 4;
+
+// unknown operators are disallowed
+// (otherwise they are no-ops with well defined cost)
+pub const NO_UNKNOWN_OPS: u32 = 8;
+
+// this constant is exported by the python binding, but when building and
```

Comment:
Consider defining it in the python bindings code, so we don't need the `dead_code` label.